pkg/frontend/adminactions/kubeactions.go

```diff
@@ -72,6 +80,28 @@ func NewKubeActions(log *logrus.Entry, env env.Interface, oc *api.OpenShiftClust
 	}, nil
 }
 
+func (k *kubeActions) KubeGetPodLogs(ctx context.Context, namespace string, name string, containerName string) ([]byte, error) {
+	opts := v1.PodLogOptions{Container: containerName}
+	if k.kubeclient.Clientset == nil {
+		err := api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "", "kubeclient's Clientset is empty in KubeActions")
+		k.log.Error(err)
+		return nil, err
+	}
+
+	r := k.kubeclient.Clientset.CoreV1().Pods(namespace).GetLogs(name, &opts)
+
+	result, err := r.Do(ctx).Raw()
+	if err != nil {
+		k.log.Error(err)
+		return nil, err
+	}
+	return result, nil
+}
+
+func (k *kubeActions) UpdateKubernetesInterface(kubeInterface kubernetes.Interface) {
```

Comment:
I don't understand what this is for -- why not just pass it in KubeGetPodLogs?

Reply:
I thought of adding it to the kubeActions struct itself .. later now realized it is not required to alter that .. 
made the changes.. please do check in the new commit.